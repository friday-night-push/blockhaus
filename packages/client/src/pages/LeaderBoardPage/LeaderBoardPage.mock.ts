export type TLeaderBoard = {
  id: number;
  name: string;
  score: number;
};

export const LEADERBOARD_MOCK: TLeaderBoard[] = [
  {
    id: 1,
    name: 'John',
    score: 100,
  },
  {
    id: 2,
    name: 'Jane',
    score: 90,
  },
  {
    id: 3,
    name: 'Jack',
    score: 80,
  },
];
