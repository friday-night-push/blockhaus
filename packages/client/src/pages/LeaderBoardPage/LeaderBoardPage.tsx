import { useEffect, useState } from 'react';

import { ClockArrowRotateLeft } from '@gravity-ui/icons';
import { Arrows3RotateLeft } from '@gravity-ui/icons';

import { Text } from '@gravity-ui/uikit';

import { Container } from 'src/components/atoms/Container';
import { Page } from 'src/components/organisms/Page';

import LeaderboardAPI from '../../services/api/leaderboard-api';

const lbApi = new LeaderboardAPI();

export const LeaderBoardPage = () => {
  const [showLb, setShowLb] = useState(false);
  const [lb, setLb] = useState<any[]>([]);

  useEffect(() => {
    const loadLeaderboard = async () => {
      await lbApi.getLeaderboard(updateLb, (err: Error) => {
        console.error(err);
        setShowLb(false);
      });
    };

    loadLeaderboard();
  }, []);

  const updateLb = (data: any) => {
    setShowLb(data.length > 0);
    let key = 1;
    const lbdata = data.map((d: any) => {
      return { ...d.data, id: key++ };
    });
    setLb(lbdata);
  };

  const getDiff = (diff: number) => {
    switch (diff) {
      case 7:
        return 'Easy';
      case 5:
        return 'Medium';
      case 3:
        return 'Hard';
      case 1:
        return 'Nightmare';
      default:
        return null;
    }
  };

  return (
    <Page title='Leaderboards' withHeader hasBackButton isFullWidth>
      <Container direction='column' alignItems='center' width='100%' gap={8} grow>
        <Container direction='column' gap={2}>
          {showLb && (
            <Container key={10000} alignItems='center' width='440px' justifyContent='space-between'>
              <div style={{ width: '200px', fontWeight: 'bold', fontSize: '20px' }}>User name</div>
              <Text variant='subheader-3'>Score</Text>
              <Text variant='subheader-3'>Diff</Text>
              <Text variant='subheader-3'>Type</Text>
            </Container>
          )}
          {showLb &&
            lb.map(user => (
              <Container key={user.id} alignItems='center' width='440px' justifyContent='space-between'>
                <div style={{ width: '200px', fontWeight: 'bold', fontSize: '20px' }}>{user.name}</div>
                <Text variant='subheader-3'>{user.score}</Text>
                <Text variant='subheader-3'>{getDiff(user.diff)}</Text>
                <Text variant='subheader-3'>{user.type == '0' ? <ClockArrowRotateLeft /> : <Arrows3RotateLeft />}</Text>
              </Container>
            ))}
          {!showLb && <Text variant='subheader-1'>No leaderboard data</Text>}
        </Container>
      </Container>
    </Page>
  );
};
