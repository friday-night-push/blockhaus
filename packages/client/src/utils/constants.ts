export const HOST = 'http://friday-night-push-blockhau-40.ya-praktikum.tech';
export const SERVER_HOST_URL = `${HOST}:3001`;
export const API_URL = `${SERVER_HOST_URL}/api/v1`;
export const YANDEX_URL = `${API_URL}/proxy`;

// OAuth
export const REDIRECT_URL = `${HOST}:3000`;

export const RESOURCE_URL = YANDEX_URL + '/resources';

export enum PAGE_ROUTES {
  MENU = '/',
  SIGN_IN = '/sign-in',
  SIGN_UP = '/sign-up',
  CHANGE_PASSWORD = '/change-password',
  RECOVER_PASSWORD = '/recover-password',
  PROFILE = '/profile',
  GAME = '/game',
  GAME_PAUSE = '/game-pause',
  GAME_RESULT = '/game-result',
  GAME_SETUP = '/game-setup',
  GAME_DIFFICULT = '/game-difficult',
  LEADER_BOARD = '/leader-board',
  ERROR_BOUNDARY = '/error-boundary',
  FORUM = '/forum',
  FORUM_TOPIC = '/forum/:topicId/:page?',
}
