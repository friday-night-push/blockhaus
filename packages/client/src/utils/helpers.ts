// Отдельный класс исключений для ESLint

export default class Helpers {
  public static Log(level: string, ...args: any): any {
    switch (level) {
      case 'INFO':
        // eslint-disable-next-line no-console
        console.info(...args);
        break;
      case 'LOG':
        // eslint-disable-next-line no-console
        console.log(...args);
        break;
      case 'ERROR':
        // eslint-disable-next-line no-console
        console.error(...args);
        break;
      default:
        break;
    }
  }
}
