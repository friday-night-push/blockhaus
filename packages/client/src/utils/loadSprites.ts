export function loadSprites(
  spritesPath: Record<string, string> = {}
): Promise<Record<string, HTMLImageElement>> {
  return Promise.all(
    Object.values(spritesPath).map(path => loadImage(path))
  ).then(sprites => {
    const keys = Object.keys(spritesPath);
    const map = keys.map((key, index) => [key, sprites[index]]);
    return Object.fromEntries(map);
  }) as Promise<Record<string, HTMLImageElement>>;
}

function loadImage(src: string): Promise<HTMLImageElement> {
  return new Promise<HTMLImageElement>((resolve, reject) => {
    const img = new Image();
    img.onload = () => resolve(img);
    img.onerror = reject;
    img.src = src;
  });
}
