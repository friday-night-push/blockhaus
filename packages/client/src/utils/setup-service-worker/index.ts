export * from './setup-service-worker';
