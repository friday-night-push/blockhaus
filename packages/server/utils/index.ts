export * from './init-postgres';
export * from './is-dev';
export * from './logger';
