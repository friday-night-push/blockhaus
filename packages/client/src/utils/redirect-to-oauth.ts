import { REDIRECT_URL } from './constants';

export const redirectToOAuth = (serviceId: string) => {
  window.location.replace(
    `https://oauth.yandex.ru/authorize?response_type=code&client_id=${serviceId}&redirect_uri=${REDIRECT_URL}`
  );
};
