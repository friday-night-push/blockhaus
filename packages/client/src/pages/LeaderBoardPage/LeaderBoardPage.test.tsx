import { render } from 'src/utils/tests';

import { LeaderBoardPage } from './LeaderBoardPage';

describe('LeaderBoardPage', () => {
  it('match snapshot', () => {
    const { container } = render(<LeaderBoardPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
