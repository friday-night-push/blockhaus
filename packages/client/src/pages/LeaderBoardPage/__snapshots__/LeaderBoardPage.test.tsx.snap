// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoardPage match snapshot 1`] = `
<section
  class="g-box g-flex"
  style="width: 100%; min-height: 100vh; column-gap: 12px; row-gap: 12px; justify-content: center;"
>
  <section
    class="g-box g-flex"
    style="width: 100%; min-height: 100vh; max-width: none; flex-direction: column; column-gap: 32px; row-gap: 32px; align-items: center; justify-content: center;"
  >
    <section
      class="g-box g-flex"
      style="width: 100%; min-height: 80px; column-gap: 12px; row-gap: 12px; align-items: center; justify-content: space-between; padding: 0px 20px;"
    >
      <button
        class="g-button g-button_view_flat g-button_size_xl g-button_pin_brick-brick"
        type="button"
      >
        <span
          class="g-button__icon g-button__icon_side_start"
        >
          <span
            class="g-button__icon-inner"
          >
            <svg
              aria-hidden="true"
              class="g-icon"
              fill="currentColor"
              height="16"
              stroke="none"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <svg
                fill="none"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14.75 8a.75.75 0 0 1-.75.75H3.81l2.72 2.72a.75.75 0 1 1-1.06 1.06l-4-4a.75.75 0 0 1 0-1.06l4-4a.75.75 0 0 1 1.06 1.06L3.81 7.25H14a.75.75 0 0 1 .75.75"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </svg>
          </span>
        </span>
        <span
          class="g-button__text"
        >
          Back
        </span>
      </button>
      <section
        class="g-box g-flex"
        style="column-gap: 12px; row-gap: 12px; align-items: center;"
      >
        <button
          class="g-button g-button_view_flat-action g-button_size_xl g-button_pin_brick-brick"
          type="button"
        >
          <span
            class="g-button__text"
          >
            Sign In
          </span>
        </button>
      </section>
    </section>
    <section
      class="g-box g-flex"
      style="width: 100%; flex-direction: column; flex-grow: 1; column-gap: 32px; row-gap: 32px; align-items: center;"
    >
      <section
        class="g-box g-flex g-flex_center-content"
        style="flex-direction: column; column-gap: 12px; row-gap: 12px;"
      >
        <span
          class="g-text g-text_variant_display-1"
        >
          Here some of the best players
        </span>
        <span
          class="g-text g-text_variant_subheader-1"
        >
          You can be one of them
        </span>
      </section>
      <section
        class="g-box g-flex"
        style="flex-direction: column; column-gap: 8px; row-gap: 8px;"
      >
        <section
          class="g-box g-flex"
          style="width: 340px; column-gap: 12px; row-gap: 12px; align-items: center; justify-content: space-between;"
        >
          <div
            class="g-user-label g-user-label_view_clear g-user-label_size_xl"
          >
            <div
              class="g-user-label__main"
            >
              <div
                class="g-user-label__avatar"
              >
                <div
                  class="g-avatar g-avatar_size_xl g-avatar_theme_normal g-avatar_view_filled"
                  role="img"
                >
                  <div
                    class="g-avatar__text"
                  >
                    JO
                  </div>
                </div>
              </div>
              <div
                class="g-user-label__text"
              >
                John
              </div>
            </div>
          </div>
          <span
            class="g-text g-text_variant_subheader-3"
          >
            100
          </span>
        </section>
        <section
          class="g-box g-flex"
          style="width: 340px; column-gap: 12px; row-gap: 12px; align-items: center; justify-content: space-between;"
        >
          <div
            class="g-user-label g-user-label_view_clear g-user-label_size_xl"
          >
            <div
              class="g-user-label__main"
            >
              <div
                class="g-user-label__avatar"
              >
                <div
                  class="g-avatar g-avatar_size_xl g-avatar_theme_normal g-avatar_view_filled"
                  role="img"
                >
                  <div
                    class="g-avatar__text"
                  >
                    JA
                  </div>
                </div>
              </div>
              <div
                class="g-user-label__text"
              >
                Jane
              </div>
            </div>
          </div>
          <span
            class="g-text g-text_variant_subheader-3"
          >
            90
          </span>
        </section>
        <section
          class="g-box g-flex"
          style="width: 340px; column-gap: 12px; row-gap: 12px; align-items: center; justify-content: space-between;"
        >
          <div
            class="g-user-label g-user-label_view_clear g-user-label_size_xl"
          >
            <div
              class="g-user-label__main"
            >
              <div
                class="g-user-label__avatar"
              >
                <div
                  class="g-avatar g-avatar_size_xl g-avatar_theme_normal g-avatar_view_filled"
                  role="img"
                >
                  <div
                    class="g-avatar__text"
                  >
                    JA
                  </div>
                </div>
              </div>
              <div
                class="g-user-label__text"
              >
                Jack
              </div>
            </div>
          </div>
          <span
            class="g-text g-text_variant_subheader-3"
          >
            80
          </span>
        </section>
      </section>
    </section>
  </section>
</section>
`;
